guide.py

```diff
@@ -0,0 +1,434 @@
+    if blink_components[0] == models.BlinkComponent.DEFAULT_COMPONENT and bug_url:
+      result = urlfetch.fetch(bug_url)
+      if result.status_code == 200:
+        soup = BeautifulSoup(result.content, 'html.parser')
+        components = soup.find_all(string=re.compile('^Blink'))
+
+        h = HTMLParser()
+        return [h.unescape(unicode(c)) for c in components]
+    return blink_components
+
+  def get(self, path, feature_id, stage_id):
+    user = users.get_current_user()
+    if user is None:
+      # Redirect to public URL for unauthenticated users.
+      return self.redirect(self.DEFAULT_URL + '/' + feature_id)
+
+    if not self._is_user_whitelisted(user):
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    template_data = {
+        }
+
+    f = models.Feature.get_by_id(long(feature_id))
+    if f is None:
+      return self.redirect(self.ADD_NEW_URL)
+
+    detail_form_class = STAGE_FORMS.get(int(stage_id), models.FeatureForm)
+
+    # Provide new or populated form to template.
+    template_data.update({
+        'feature': f,
+        'feature_id': f.key().id,
+        'feature_form': detail_form_class(f.format_for_edit()),
+    })
+
+    if self.LAUNCH_PARAM in self.request.params:
+      template_data[self.LAUNCH_PARAM] = True
+    if self.INTENT_PARAM in self.request.params:
+      template_data[self.INTENT_PARAM] = True
+
+    self._add_common_template_values(template_data)
+
+    self.render(data=template_data, template_path=os.path.join(path + '.html'))
+
+  def post(self, path, feature_id, stage_id):
+    user = users.get_current_user()
+    if user is None or (user and not self._is_user_whitelisted(user)):
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    spec_link = self.__FullQualifyLink('spec_link')
+
+    explainer_links = self.request.get('explainer_links') or []
+    if explainer_links:
+      explainer_links = filter(bool, [x.strip() for x in re.split('\\r?\\n', explainer_links)])
+
+    bug_url = self.__FullQualifyLink('bug_url')
+    intent_to_implement_url = self.__FullQualifyLink('intent_to_implement_url')
+    origin_trial_feedback_url = self.__FullQualifyLink('origin_trial_feedback_url')
+
+    ff_views_link = self.__FullQualifyLink('ff_views_link')
+    ie_views_link = self.__FullQualifyLink('ie_views_link')
+    safari_views_link = self.__FullQualifyLink('safari_views_link')
+    web_dev_views_link = self.__FullQualifyLink('web_dev_views_link')
+
+    # Cast incoming milestones to ints.
+    shipped_milestone = self.__ToInt('shipped_milestone')
+    shipped_android_milestone = self.__ToInt('shipped_android_milestone')
+    shipped_ios_milestone = self.__ToInt('shipped_ios_milestone')
+    shipped_webview_milestone = self.__ToInt('shipped_webview_milestone')
+    shipped_opera_milestone = self.__ToInt('shipped_opera_milestone')
+    shipped_opera_android_milestone = self.__ToInt('shipped_opera_android_milestone')
```

Comment:
Do all milestones map to integers (e.g., Android letter releases)?

Reply:
I believe they do for all Chromium forks. In any case, MDN has compiled [a pretty complete list](https://github.com/mdn/browser-compat-data/tree/master/browsers) of known version numbers for major browsers.

Reply:
Thanks for the MDN reference, that clarified my Android question.

This comment then becomes more of a nit.  I see a recent example where Safari did use a minor release (13.1).  I know it's come up in Chrome at least once (4.1 was the last that I recall). 
From that perspective, there really isn't anything stopping the various release teams from changing their versioning/ naming conventions.  

I suspect it's not worth a lot of code complexity, but moving away from int (to something like string) could afford value for future proofing.

Reply:
Good points.  This is existing code that relates to the existing database schema, so it would take some work to change.   I added a TODO for that.